File: packages/expo-location/android/src/main/java/expo/modules/location/LocationRequestCallbacks.java
package expo.modules.location;

import android.location.Location;

import expo.modules.core.errors.CodedException;

abstract class LocationRequestCallbacks {
  public void onLocationChanged(Location location) {}
  public void onLocationError(CodedException throwable) {}
  public void onRequestSuccess() {}
  public void onRequestFailed(CodedException throwable) {}
}

File: packages/expo-location/android/src/main/java/expo/modules/location/LocationRequestCallbacks.kt
package expo.modules.location

import android.location.Location
import expo.modules.kotlin.exception.CodedException

interface LocationRequestCallbacks {
  fun onLocationChanged(location: Location) {}
  fun onLocationError(cause: CodedException) {}
  fun onRequestSuccess() {}
  fun onRequestFailed(cause: CodedException) {}
}
